src/core/geometry/qgsrectangle.h

```diff
@@ -165,16 +165,24 @@ class CORE_EXPORT QgsRectangle
      */
     void setYMaximum( double y ) SIP_HOLDGIL { mYmax = y; }
 
+    /**
+     * Mark a rectangle as being null.
+     */
+    void setNull() SIP_HOLDGIL
+    {
+      mXmin = mYmin = std::numeric_limits<double>::max();
+      mXmax = mYmax = -std::numeric_limits<double>::max();
```

Comment:
Nit: I'd slightly prefer `std::numeric_limits<double>::lowest()` if available (C++11).

Reply:
is there an ::highest() too ?

Reply:
That's called `max` :slightly_smiling_face:.

Reply:
in any case @benoitdm-oslandia  [suggested to use NaN](https://github.com/qgis/QGIS/pull/54646#discussion_r1330107062) so this is still I think a detail that shouldn't be handled in this PR which is purely for the API part.

Reply:
I guess it depends on how exciting you want calling `center` on a null `QgsRectangle` or doing an `intersect` between a null and a non-null one to be.

Reply:
I would change semantic of "isEmpty()" to mean "it represents an no space at all" (what isNull() currently means) and drop isNull/setNull all togheter. It would probably be the easiest to explain. I dubt any user is relying on "isEmpty()" meaning "area == 0"